fscache.go

```diff
@@ -175,105 +181,66 @@ func (c *cache) Clean() error {
 }
 
 type cachedFile struct {
-	fs   FileSystem
-	name string
-	cnt  int64
-	grp  sync.WaitGroup
-	w    io.WriteCloser
-	b    *broadcaster
+	stream *stream.Stream
+	cnt    int64
 }
 
 func (c *cache) newFile(name string) (*cachedFile, error) {
-	f, err := c.fs.Create(name)
+	s, err := stream.NewStream(name, c.fs)
 	if err != nil {
 		return nil, err
 	}
 	cf := &cachedFile{
-		fs:   c.fs,
-		name: f.Name(),
-		w:    f,
-		b:    newBroadcaster(),
+		stream: s,
 	}
-	cf.grp.Add(1)
 	atomic.AddInt64(&cf.cnt, 1)
 	return cf, nil
 }
 
 func (c *cache) oldFile(name string) *cachedFile {
-	b := newBroadcaster()
-	b.Close()
-	return &cachedFile{
-		fs:   c.fs,
-		name: name,
-		b:    b,
+	s, _ := stream.NewStream(name, c.fs)
```

Comment:
Ah good catch. However, I think we can solve this without touch stream.

I'm looking into it at the moment. But we might be able to just return the file in this case with FileSystem.Open(). Stream is really only a wrapper over a file which provides synchronization between concurrent reads/writes. If the file is completely written (like with oldFile) we just need to serve this file instead.